src/assets/scripts/client/base/BaseModel.js

```diff
@@ -45,4 +45,27 @@ export default class BaseModel {
     reset() {
         throw new TypeError('BaseModel#reset method must be implemented by the class extending BaseModel');
     }
+
+    /**
+     * This will verify if the given argument is a string and will return a modified pre-value for the _id
+     *
+     * @for BaseModel
+     * @method veriftyModelName
+     * @param {string / Object}
+     */
+    verifyModelName(modelName) {
+        if (typeof modelName === 'string') {
+            return `${modelName}-`;
+        }
+
+        // This will give more infomation on airpot names and runways to help identify if there is an issue.
+        if (modelName.name === undefined && typeof modelName === 'object') {
```

Comment:
I see what you're doing there.  That actually isn't the intent.  I may not have explained the terms here very well (and they do collide a little, which is confusing).

This class, and all classes inheriting it, has an `_id` property.  Without any input from us, this `_id` will be a string of numbers that increments every time this class is instantiated.  so you'll end up with `_id`s that look like: `this._id = 3432` and so on.  

My intent for this is for an inheriting class to have the ability to send an optional sting that can be added to the `_id` that helps differentiate each instance.  so say we have the `FixModel`, which inherits the `BaseModel`.  In the `FixModel` constructor, we would call super with that optional `_id` string : `super('FixModel')` which would then be added to that id.  this would give us an `_id` of `FixModel-3432`.

I think the _model_ you are thinking of here is actually the `AircraftTypeDefinitionModel` or the `AirlineModel`, those are different things entirely.  So this block can go away.  It was a good thought, but thats not what I'm looking for here.